### DISCUSSION TITLE
Permission Issues with Docker in Concourse Pipeline on macOS (M3 Chip)

### DISCUSSION THREAD

**Discussion by johna77** [concourse/concourse#9025] [Help & Support] [answered]
I'm running into permission issues when executing a Concourse pipeline on macOS (M3 chip) with Docker Desktop installed. The same pipeline works perfectly on Windows, but on macOS, I encounter the following error:
`selected worker: [worker-id].    
mount: permission denied (are you root?).    
FATA[0000] failed to start buildkitd: setup cgroups: exit status 1        
FATA[0000] failed to run task: exit status 1           
`
I have enabled privileged: true in my pipeline configuration to grant elevated permissions, and I've verified that Docker Desktop is up-to-date.

I am using docker-compose.yaml from official doc - 
`houdini`
` privileged: true`

If anyone has experience running Concourse on macOS with Docker Desktop or has faced similar permission-related issues, I'd appreciate any insights or solutions!
<img width="896" alt="image" src="https://github.com/user-attachments/assets/c6cd2d26-8740-4fad-bb4c-b5609a497d65">

**Comment by taylorsilva** [comment, marked as answer]
This is an arm64 vs amd64 issue.

On Windows you're likely running Concourse on the Docker linux VM, so all the Concourse defaults probably work. Runtime can be containerd and tasks like that one you're trying to run there work without issue.

On macOS with an M-series chip, you're running on arm64. Docker will _try_ and run amd64 images on arm64 architectures, but when you're trying to do container stuff things break down.

In general I would say it's not currently possible to 100% run Concourse on any arm-based CPU.

This issue will eventually resolve this: https://github.com/concourse/concourse/issues/1379
